### PR TITLE
Pip Packaging

### LINKED ISSUE #52: PIP Packaging Support
- [x] Update necessary files to properly support installation with pip.
- [x] Update the build system and CICD to build and release pip packages on PyPi.

### PR DESCRIPTION
**Pull request by Morpheus636**
## Describe Your Changes:
- Reorganized files to enable building for install via pip
- Wrote a GitHub action to release builds via pypi

### Linked Issues:
Please use Closing Keywords to link any related issues. i.e:
Closes #52

### Category
- [ ] This is a Minor Change which does not fit any other category.
- [ ] This is a Bugfix
- [x] This is a Feature Addition
- [x] This is a CI Change
- [ ] This is a Documentation-Only Change (do not check any of the above boxes if this box is checked)
### Breaking/Non Breaking
- [x] This is a Breaking Change
- [ ] This is a Non-Breaking Change

#### Explanation
This potentially breaks API compatibility, as it changes the location of the entire `zeal` package within `src`. This does not affect CLI or GUI usage.

## Copyright Assignment
- [ ] I have filled out the copyright assignment at https://copyright.morpheus636.com for this pull request.
- [x] I **am** Morpheus636 so I do not need to fill out the copyright assignment.

#### Why do I require a copyright assignment?
<details>
<summary> Click to expand </summary>
I require a copyright assignment for changes to my projects for the same reason as the Free Software Foundation, and they do a much better job of explaining it in their post at https://www.gnu.org/licenses/why-assign.en.html than I could ever do.

Please email any questions about the copyright assignment to COPYRIGHT@morpheus636.com
</details>

### CODE REVIEW

**Comment by Morpheus636** [comment]
@alkuzad Any chance you could look this over for me?

**Comment by Morpheus636** [comment]
Still fixing some stuff with tests and pyinstaller though so no rush.

**Comment by Morpheus636** [comment]
Alright tests are updated but pyinstaller is still erroring out for some reason. Something about setuptools.

**Comment by alkuzad** [comment]
@Morpheus636 for this to work you have to clean a bit:

Make all imports absolute using package name, not relative import and import path manipulation. Package produced won't work with relative. `src` in import has to go. 
This is the setup.cfg snippet for using directly package, without src:
```
[options]
include_package_data = True
package_dir=
    =src
packages=find:

[options.packages.find]
where=src
exclude =
    tests
    *.tests
    *.tests.*
    tests.*
```

As for error - you removed version.py so the update script cannot work properly :)

**Comment by Morpheus636** [comment]
> As for error - you removed version.py so the update script cannot work properly :)
Yup, on my list

> src in import has to go.
Import path manipulation and src in import paths are only in tests. The code itself is all relative. Can you explain why relative won't work? Installing via pip + git in a test environment worked just fine for me.

**Comment by alkuzad** [comment]
Ok, I made some refreshement of imports and yeah, the "explicit relative" which we use here with the `.` in before should behave the same way as absolute does. I just had so many problems with relative in the past that stuck to absolute for all cases and never really looked into changes around relative imports - https://peps.python.org/pep-0328/.

Still the src prefix looks bad and it should be gone with the changes to build system :)

**Comment by Morpheus636** [comment]
@alkuzad Can you take a look at this error for me? Did I screw up something in the tests?

**Comment by Morpheus636** [comment]
@alkuzad Can you review the changes?

File: .github/workflows/pypi_build.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        uses: Gr1N/setup-poetry@v7
+
+      - name: Install Packages
+        run: poetry install --no-dev
+
+      - name: Build
+        run: poetry build
+
+      - name: Publish
+        run: poetry publish -u ${{ secrets.PYPI_USER}} -p ${{ secrets.PYPI_PASS}}
```
Comment (alkuzad) [inline .github/workflows/pypi_build.yml:29]: consider masking https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#masking-a-value-in-log

File: .github/workflows/pypi_build.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
```
Comment (alkuzad) [inline .github/workflows/pypi_build.yml:17]: in tests we have 3.10, inconsistent or consider just using tox in matrix

Comment (Morpheus636) [reply .github/workflows/pypi_build.yml:17]: Oops! I just forgot to change it when I copied it from an old project. Thanks for catching it.

File: .github/workflows/pypi_build.yml (hunk 1 above)
Comment (Morpheus636) [reply .github/workflows/pypi_build.yml:29]: Secrets are redacted from logging by default. https://docs.github.com/en/actions/security-guides/encrypted-secrets#accessing-your-secrets

**Review by alkuzad — APPROVED** [review]

**Pull request merged**